core/capabilities/ccip/ccipevm/pluginconfig.go

```diff
@@ -0,0 +1,35 @@
+package ccipevm
+
+import (
+	chainsel "github.com/smartcontractkit/chain-selectors"
+	ccipcommon "github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/common"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/ocrimpls"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+const defaultCommitGasLimit = 500_000
+
+// PluginConfig is a struct that contains the configuration for a plugin.
+type PluginConfig struct {
+	extraDataCodec ccipcommon.ExtraDataCodec
+}
+
+// NewPluginConfig returns a new PluginConfig.
+func NewPluginConfig(extraDataCodec ccipcommon.ExtraDataCodec) *PluginConfig {
+	return &PluginConfig{
+		extraDataCodec: extraDataCodec,
+	}
+}
```

Comment:
Not sure why do we need this local `PluginConfig` type exposed, as you can keep it internal and just return the `OffChainPluginConfig` interface type from the constructor. That interface type should also be renamed as `PluginConfigProvider` as the code currently suggests.